### PR TITLE
Add type hints

### PR DESCRIPTION
**Pull request by araffin**
Note: this is WIP, I need to add more file later

### CODE REVIEW

**Comment by tawnkramer** [comment]
What python versions work w these changes?

**Comment by araffin** [comment]
> What python versions work w these changes?

python 3.5+ (but python 3.5 is no longer maintained)

**Comment by tawnkramer** [comment]
please let me know when this is ready to be merged.

**Comment by araffin** [comment]
@Heavy02011 I renamed some config variables of the lidar to be snake case as the rest of the repo.
@tawnkramer I think it is ready to review (I will do a final check after lunch)

**Review by tawnkramer — COMMENTED** [review]
Looks great. Sorry so long to merge

**Comment by araffin** [comment]
> Looks great. Sorry so long to merge

Thanks, I need to fix conflicts first...
And please tell me if you agree with the renaming of config variables for the lidar?

Pinging @Maximellerbach too

**Comment by Maximellerbach** [comment]
Hey @araffin ! Renaming the lidar config variables would mean to also rename them in the sim, so this PR would have to be merge at the same time as a new sim release. 
What I suggest is that we create a new PR dedicated to just the renaming of those variables, so that it doesn't block the merging of this one.

**Comment by araffin** [comment]
> What I suggest is that we create a new PR dedicated to just the renaming of those variables, so that it doesn't block the merging of this one.

I am actually keep the sim names when sending data https://github.com/tawnkramer/gym-donkeycar/pull/59/files#diff-193e6cc0eb5696e544ffd3252f95ed73b485eefe7e52387c97e017ed8eb33ff4R585
The issue is more that this is a breaking change for people using the donkey gym package, and I wanted to know is this was ok or should I write backward compatible code that warn users too?

**Comment by Maximellerbach** [comment]
ohhh, I see sorry. In this case I guess a little warning reminding the valid arguments would be ideal.

**Comment by araffin** [comment]
> ohhh, I see sorry. In this case I guess a little warning reminding the valid arguments would be ideal.

Ready for final review ;) (please test it if you can)

File: HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,13 @@
 History
 =======
 
+1.2.0 (WIP)
```
Comment (Maximellerbach) [inline HISTORY.rst:5]: remove WIP

**Review by Maximellerbach — APPROVED** [review]

File: HISTORY.rst (hunk 1 above)
Comment (Maximellerbach) [reply HISTORY.rst:5]: other than that, I approve those changes 🙂. Really good work, thank you !

**Comment by Maximellerbach** [comment]
I will merge this along with the new version of the sim !

**Pull request merged**